xbmc/cores/AudioEngine/AESinkFactory.h

```diff
@@ -42,15 +42,27 @@ struct AESinkRegEntry
   Cleanup cleanupFunc;
 };
 
+struct AESinkDevice
+{
+  std::string driver;
+  std::string name;
+  std::string friendlyName;
+
+  bool IsSameDeviceAs(const AESinkDevice& d) const
```

Comment:
Can this just be `operator==` ?

Reply:
It was until I made a comment in the opposite direction :-) because the code using it looked like
```
if (saveChanges && oldDevice == newDevice)
  SaveSettings(newDevice)
```
which is strange (to me).  It's not an equality of all members if you look closely. Instances with different name/friendly name members could be considered equal.
It's also possible this condition doesn't make sense on platforms other than Windows, maybe we need a platform or sink-specific condition instead.
But I'll go with the consensus.

Reply:
I'm not sure what you mean? I was saying to overload the operator== so that we can define the evaluation.